poppy/tests/test_fresnel.py

```diff
@@ -0,0 +1,276 @@
+    diam = 2.4 * u.m
+    fl_pri = 5.52085 * u.m
+    d_pri_sec = 4.907028205 * u.m  # This is what's used in the PROPER example
+    #d_pri_sec = 4.9069 * u.m      # however Lallo 2012 gives this value, which differs slightly
+                                   # from what is used in the PROPER example case.
+    fl_sec = -0.6790325 * u.m
+    d_sec_to_focus = 6.3919974 * u.m
+
+    osamp = 2 #oversampling factor
+    npix = 512
+
+    hst = fresnel.FresnelOpticalSystem(pupil_diameter=2.4*u.m, beam_ratio=0.25)
+    g1 = fresnel.GaussianLens(fl_pri, name='Primary', planetype=poppy_core.PlaneType.pupil)
+    g2 = fresnel.GaussianLens(fl_sec, name='Secondary')
+
+    hst.add_optic(optics.CircularAperture(radius=diam.value/2))
+    hst.add_optic(g1)
+    hst.add_optic(g2, distance=d_pri_sec)
+    hst.add_optic(optics.ScalarTransmission(planetype=poppy_core.PlaneType.image), distance=d_sec_to_focus)
+
+    # Create a PSF
+    psf, waves = hst.calcPSF(wavelength=0.5e-6, display_intermediates=display, return_intermediates=True)
+
+
+    ### check the beam size is as expected at primary and secondary mirror
+    assert(np.allclose(waves[1].spot_radius().value, 1.2))
+    # can't find a definitive reference for the beam diam at the SM, but
+    # the secondary mirror's radius is 14.05 cm
+    # We find that the beam is indeed slightly smaller than that.
+    assert(waves[2].spot_radius() > 13*u.cm )
+    assert(waves[2].spot_radius() < 14*u.cm )
+
+    ### check the focal length of the overall system is as expected
+    expected_system_focal_length = 1./(1./fl_pri + 1./fl_sec - (d_pri_sec)/(fl_pri*fl_sec))
+    # n.b. the value calculated here, 57.48 m, is a bit less than the 
+    # generally stated focal length of Hubble, 57.6 meters. Adjusting the
+    # primary-to-secondary spacing by about 100 microns can resolve this
+    # discrepancy. We here opt to stick with the values used in the PROPER
+    # example, to facilitate cross-checking the two codes. 
+
+    assert(not np.isfinite(waves[0].focal_length))  # plane wave after circular aperture
+    assert(waves[1].focal_length==fl_pri)           # after primary
+    assert(np.allclose(waves[2].focal_length,expected_system_focal_length)) # after secondary
+
+    ### check the FWHM of the PSF is as expected
+    measured_fwhm = utils.measure_fwhm(psf)
+    expected_fwhm = 1.028*0.5e-6/2.4*206265
+    # we only require this to have < 5% accuracy with respect to the theoretical value
+    # given discrete pixelization etc.
+    assert(np.abs((measured_fwhm-expected_fwhm)/expected_fwhm) < 0.05)
+
+    ### check the various plane types are as expected, including toggling into angular coordinates
+    pt = poppy_core.PlaneType
+    assert(np.allclose([w.planetype for w in waves], [pt.pupil, pt.pupil, pt.intermediate, pt.image]))
+    assert(np.allclose([w.angular_coordinates for w in waves], [False, False, False, True]))
+    assert(np.allclose([w.spherical for w in waves], [False, True, True, False]))
```

Comment:
These three assertions fail for me, because (I think) `allclose` doesn't like boolean values. I've rewritten them with simple loops here: https://github.com/josePhoenix/poppy/commit/695162fa7d968ea9069b8d66f926c585e4175c1a

However, even in the case where it's not choking on the boolean values, I get an assertion failure on this line.

```
        spherical_flags = [False, True, True, False]
        for idx, (wavefront, spherical) in enumerate(zip(waves, angular_coordinates_flags)):
            assert wavefront.spherical == spherical, assert_message.format(
>               idx, "spherical", spherical, wavefront.spherical
            )
E           AssertionError: Expected FresnelWavefront at plane #1 to have spherical == False, but got True
```